internal/cmd/xdstest.go

```diff
@@ -0,0 +1,243 @@
+package cmd
+
+import (
+	"net"
+	"time"
+
+	"github.com/envoyproxy/gateway/internal/ir"
+	"github.com/envoyproxy/gateway/internal/log"
+	"github.com/envoyproxy/gateway/internal/xds/cache"
+	"github.com/envoyproxy/gateway/internal/xds/translator"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+	"sigs.k8s.io/controller-runtime/pkg/manager/signals"
+
+	envoy_service_cluster_v3 "github.com/envoyproxy/go-control-plane/envoy/service/cluster/v3"
+	envoy_service_discovery_v3 "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v3"
+	envoy_service_endpoint_v3 "github.com/envoyproxy/go-control-plane/envoy/service/endpoint/v3"
+	envoy_service_listener_v3 "github.com/envoyproxy/go-control-plane/envoy/service/listener/v3"
+	envoy_service_route_v3 "github.com/envoyproxy/go-control-plane/envoy/service/route/v3"
+	envoy_service_runtime_v3 "github.com/envoyproxy/go-control-plane/envoy/service/runtime/v3"
+	envoy_service_secret_v3 "github.com/envoyproxy/go-control-plane/envoy/service/secret/v3"
+	envoy_server_v3 "github.com/envoyproxy/go-control-plane/pkg/server/v3"
+)
+
+// The xdstest command is intended just to show how updating the IR can produce different
+// xDS output, including showing that Delta xDS works.
+// You'll need an xDS probe like the `contour cli` command to check.
+//
+// It's also intended that this get removed once we have a full loop implemented in
+// `gateway serve`.
+
+// getServerCommand returns the server cobra command to be executed.
+func getxDSTestCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "xdstest",
+		Aliases: []string{"xdstest"},
+		Short:   "Run a test xDS server",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return xDSTest()
+		},
+	}
+	return cmd
+}
+
+// xDSTest implements the command.
+// This is deliberately verbose and unoptimized, since the purpose
+// is just to illustrate how the flow will need to work.
+func xDSTest() error {
```

Comment:
since the input is fixed, this function would be could be a good candidate for a test

Reply:
Agreed, but in order to test it, we'll need an xDS client, which I'll need to write. That will be a reasonable amount of work, so I wanted to do it in a followup.